BeTime/BeTimeTests/FetchWeatherUseCaseTests.swift

```diff
@@ -7,29 +7,119 @@
 
 import XCTest
 
+import RxSwift
+
+@testable import BeTime
+
 final class FetchWeatherUseCaseTests: XCTestCase {
 
-    override func setUpWithError() throws {
-        // Put setup code here. This method is called before the invocation of each test method in the class.
-    }
-
-    override func tearDownWithError() throws {
-        // Put teardown code here. This method is called after the invocation of each test method in the class.
-    }
-
-    func testExample() throws {
-        // This is an example of a functional test case.
-        // Use XCTAssert and related functions to verify your tests produce the correct results.
-        // Any test you write for XCTest can be annotated as throws and async.
-        // Mark your test throws to produce an unexpected failure when your test encounters an uncaught error.
-        // Mark your test async to allow awaiting for asynchronous code to complete. Check the results with assertions afterwards.
-    }
-
-    func testPerformanceExample() throws {
-        // This is an example of a performance test case.
-        self.measure {
-            // Put the code you want to measure the time of here.
-        }
-    }
+  func sut(repository: WeatherDataRepositoryProtocol) -> FetchWeatherUseCaseImpl {
+    return FetchWeatherUseCaseImpl(
+      weatherRepository: repository
+    )
+  }
+
+  func test_Repository의_requestWeatherData가_올바르게_호출됐는지_확인하는_테스트() {
+
+    // Given
+    let repositoryMock = WeatherDataRepositoryMock()
+    let usecase = FetchWeatherUseCaseImpl(
+      weatherRepository: repositoryMock
+    )
```

Comment:
given에서 테스트 대상객체를 명확히 하는 것이 좋을 것 같아요.
```suggestion
    let sut = sut()
```